### PR TITLE
fixed shape mismatch in AdaptiveImageLossFunction

### PR DESCRIPTION
**Pull request by rvarun7777**

### CODE REVIEW

**Comment by jonbarron** [comment]
There are unit tests that need to be updated as well.

**Comment by rvarun7777** [comment]
I could not find an issue in your unit tests.

**Comment by rvarun7777** [comment]
Hi, unfortunately we don't have a code release for the monocular depth estimation experiments of the paper (though that code is in TF anyways so it likely isn't what you're looking for). I believe that there are Pytorch implementations of SFMLearner on Github, and using this loss should be straightforward: just delete the existing multiscale photometric loss and the smoothness term and add in AdaptiveImageLossFunction on the full-res image with:
scale_lo=0.01
scale_init=0.01
and default settings for the rest and it should work (you may need to fiddle with the value of wavelet_scale_base).

Do you suggest any other changes apart from this? I am actually testing this :)

**Comment by rvarun7777** [comment]
Does it not go well with SSIM and Smoothness loss?  You have suggested to delete the existing multiscale photometric loss and the smoothness term.

**Comment by rvarun7777** [comment]
Tested your loss on YUV space on monodepth2 project. It just drives the final loss below 0.#
AdaptiveImageLossFunction((3, 640, 192), torch.float32, "cuda:0", color_space='YUV', scale_lo=0.01, scale_init=0.01)
These were the initial setting used.
Final loss = 0.85 Adaptive Loss + 0.15 SSIM + 0.01 Smoothness

**Comment by jonbarron** [comment]
I patched your PR and ran nosetests from the root directory, and it causes
the testImageLossfun* tests to fail. Those tests need to also be rewritten
with the same NHWC-> NCHW dimension reordering.
